src/test/java/ru/training/at/hw2/ex1/JdiPageSeleniumFirstTest.java

```diff
@@ -0,0 +1,194 @@
+                        By.cssSelector(("span#user-name"))));
+        softAssert.assertTrue(logged.isDisplayed());
+        softAssert.assertEquals(logged.getText(), "ROMAN IOVLEV");
+
+        //5. Assert that there are 4 items on the header and they correspond to expected
+        List<WebElement> headerElements = webDriverWait
+                .until(webDriver -> webDriver.findElements(
+                By.xpath("//header/div/nav/ul[1]/li")));
+        //By.cssSelector("header > div > nav > ul > li")));
+        int expectedNoOfItems = 4;
+        softAssert.assertEquals(headerElements.size(), expectedNoOfItems);
+        List<String> expectedHeaderFields =
+                Arrays.asList("HOME", "CONTACT FORM", "SERVICE", "METALS & COLORS");
+        for (int i = 0; i < headerElements.size(); i++) {
+            softAssert.assertEquals(headerElements.get(i).getText(),
+                    expectedHeaderFields.get(i));
+        }
+
+        //6. Assert that there are 4 images on the Index Page and they are displayed
+        List<WebElement> icons = webDriverWait
+                .until(webDriver -> webDriver.findElements(
+                By.cssSelector("div.benefit-icon")));
+        int expectedNoOfIcons = 4;
+        softAssert.assertEquals(icons.size(), expectedNoOfIcons);
+        for (WebElement e : icons) {
+            softAssert.assertTrue(e.isDisplayed());
+        }
+
+        //7. Assert that there are 4 texts on the Index Page and they have proper texts
+        List<WebElement> iconsTexts = webDriverWait
+                .until(webDriver -> webDriver.findElements(
+                        By.cssSelector("span.benefit-txt")));
+        int expectedNoOfTexts = 4;
+        softAssert.assertEquals(iconsTexts.size(), expectedNoOfTexts);
+        List<String> expectedTexts = Arrays.asList(
+                "To include good practices\nand ideas from successful\nEPAM project",
+                "To be flexible and\ncustomizable",
+                "To be multiplatform",
+                "Already have good base\n(about 20 internal and\nsome external projects),"
+                        + "\nwish to get more…");
+        for (int i = 0; i < iconsTexts.size(); i++) {
+            softAssert.assertTrue(iconsTexts.get(i).isDisplayed());
+            softAssert.assertEquals(iconsTexts.get(i).getText(),
+                    expectedTexts.get(i));
+        }
+
+        //additional. Assert texts of the main headers
+        WebElement headerText1 = webDriverWait
+                .until(webDriver -> webDriver.findElement(
+                        By.cssSelector(("h3.main-title:first-child"))));
+        String expectedHeader3 = "EPAM FRAMEWORK WISHES…";
+        assertEquals(headerText1.getText(), expectedHeader3);
+
+        WebElement headerText2 = webDriverWait
+                .until(webDriver -> webDriver.findElement(
+                        By.xpath(("//p[@name='jdi-text'][1]"))));
```

Comment:
Please do not use indexes in the paths. Logic of them could be broken, if element will be added or deleted before

Reply:
changed the locator